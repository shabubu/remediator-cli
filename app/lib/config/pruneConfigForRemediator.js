import { ALL_REMEDIATOR_CONFIG_KEYS } from 'app/constants';

/**
 * Takes in remediator-cli config and returns a config for remediator.
 * @param   {Object} config Remediator-cli config object.
 * @returns {Object}        Config with remediator keys and values only.
 */
export default function pruneConfigForRemediator(config) {
  const remediatorConfig = {};

  ALL_REMEDIATOR_CONFIG_KEYS.forEach((configKey) => {
    if (config[configKey] !== undefined) {
      remediatorConfig[configKey] = config[configKey];
    }
  });

  return remediatorConfig;
}
