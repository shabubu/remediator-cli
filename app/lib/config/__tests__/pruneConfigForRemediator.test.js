import {
  BATCH_SIZE_KEY,
  FORMAT_KEY,
  MODE_KEY,
  OUTPUT_DIRECTORY_KEY,
  RECURSIVE_KEY,
  SKIP_ERRORS_KEY,
  SOURCE_DIRECTORIES_KEY,
  VERBOSITY_KEY,
} from 'app/constants';
import pruneConfigForRemediator from 'app/lib/config/pruneConfigForRemediator';

describe('app/lib/config/pruneConfigForRemediator', () => {
  test('should return remediator config without cli config or undefined keys', () => {
    const remediatorCliConfig = {
      [BATCH_SIZE_KEY]: '2',
      [FORMAT_KEY]: ':foo:',
      [MODE_KEY]: 'copy',
      [OUTPUT_DIRECTORY_KEY]: './foo',
      [RECURSIVE_KEY]: undefined,
      [SKIP_ERRORS_KEY]: false,
      [SOURCE_DIRECTORIES_KEY]: [
        './bar',
        './foobar',
      ],
      [VERBOSITY_KEY]: true,
    };
    const result = pruneConfigForRemediator(remediatorCliConfig);

    expect.assertions(1);
    expect(result).toMatchSnapshot();
  });
});
