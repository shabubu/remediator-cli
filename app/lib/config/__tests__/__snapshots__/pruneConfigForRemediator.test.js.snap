// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/lib/config/pruneConfigForRemediator should return remediator config without cli config or undefined keys 1`] = `
Object {
  "batchSize": "2",
  "format": ":foo:",
  "mode": "copy",
  "output": "./foo",
  "skipErrors": false,
  "source": Array [
    "./bar",
    "./foobar",
  ],
}
`;
